frontend/src/components/UserSearch.vue

```diff
@@ -216,68 +245,105 @@ export default {
       this.$router.push({ name: "UserCreate" });
     },
     searchUser: function(queryParameters) {
+      const maxSearch = app_config.config.max_search
+                          ? app_config.config.max_search
+                          : (this.maxResults * 10);
       this.userSearchLoadingStatus = true;
-      let roleId = this.roleInput.trim();
-
+      let isSearchByRole = this.advancedSearchSelected
+        && this.selectedRoles.length > 0;
+      
       UsersRepository.get(
-        "?briefRepresentation=false&first=0&max=300" + queryParameters
+        `?briefRepresentation=false&first=0&max=${maxSearch}` + queryParameters
       )
         .then(response => {
           let results = response.data;
-          if (this.advancedSearchSelected && roleId !== "") {
-            this.findUserIdsInRole(roleId)
-              .then(roleSearchResults =>
-                this.searchResults = results.filter(
-                  user => roleSearchResults.includes(user.id)
-                )
-              );
+          if (isSearchByRole) {
+            this.filterUsersByRole(results, maxSearch)
+              .then(filteredResults => this.setSearchResults(filteredResults))
+              .finally(() => (this.userSearchLoadingStatus = false));
           }
           else {
-            this.searchResults = results;
+            this.setSearchResults(results);
           }
         })
         .catch(error => {
           this.handleError("User search failed", error);
         })
-        .finally(() => (this.userSearchLoadingStatus = false));
+        .finally(() => {
+          if (!isSearchByRole) {
+            this.userSearchLoadingStatus = false;
+          }
+        });
```

Comment:
I am not a Javascript expert, so please beware of what I'm about to propose. I think this code can be cleaned up using `async/await` instead of Promises. In my quick test, the following code works:

```javascript
searchUser: async function(queryParameters) {
  this.$store.commit("alert/dismissAlert");
  const maxSearch = app_config.config.max_search
                      ? app_config.config.max_search
                      : (this.maxResults * 10);
  this.userSearchLoadingStatus = true;
  let isSearchByRole = this.advancedSearchSelected
    && this.selectedRoles.length > 0;
  try {
    let results = (await UsersRepository.get(`?briefRepresentation=false&first=0&max=${maxSearch}` + queryParameters)).data;
    if (isSearchByRole) {
      results = await this.filterUsersByRole(results, maxSearch);
    }
    this.setSearchResults(results);
  } catch (e) {
    this.handleError("User search failed", e);
  } finally {
    this.userSearchLoadingStatus = false;
  }
}
```

While the code you've written is short enough that it's easy to understand, please consider this alternative.

Reply:
Addressed by https://github.com/bcgov/moh-keycloak-user-management/pull/61/commits/f80cefb45f4fc09a273ef5d12519b050f21e1e52